lib/model/CEventRatePopulationModel.cc

```diff
@@ -466,6 +466,10 @@ void CEventRatePopulationModel::sample(core_t::TTime startTime,
                     .trendWeights(attribute.second.s_Weights)
                     .priorWeights(attribute.second.s_Weights);
                 maths::CModel* model{this->model(feature, cid)};
+                if (model == nullptr) {
+                    LOG_TRACE(<< "Model unexpectedly null");
```

Comment:
In contrast to the `featureData` methods, the `model` methods _can_ returning `nullptr` without logging anything.  That makes me wonder if we should log at the `ERROR` level in these places (6 in this PR) and include `feature` and `cid` in the error messages where `model` might return `nullptr` are genuinely unexpected?  What do you think @tveasey?

Reply:
I think it should never happen. The only thing I wonder about is whether we don't create a model for say hitting memory constraints, but we still have a value for it in the bucket: I think these also get removed, but will double check.